### PR TITLE
🌱 Align docker build to CAPI

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Align docker-build (Dockerfiles, .dockerignore, make targets) to what we are doing in CAPI, use the same build targets for E2E

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -15,13 +15,23 @@
...
+# https://github.com/hadolint/hadolint/wiki/DL3006
+# hadolint ignore=DL3006
+FROM ${builder_image} as builder
 WORKDIR /workspace
 
-# Run this with docker build --build_arg $(go env GOPROXY) to override the goproxy
+# Run this with docker build --build-arg goproxy=$(go env GOPROXY) to override the goproxy
 ARG goproxy=https://proxy.golang.org
-ENV GOPROXY=${goproxy}
+# Run this with docker build --build-arg package=./test/infrastructure/vcsim
```
Comment (chrischdi) [inline Dockerfile:33]: Doesn't vcsim have its own Dockerfile further below?

File: .dockerignore [hunk 2]
```diff
@@ -1,22 +1,39 @@
...
+templates/
 **/*.md
-*.test
-cluster-api-provider-vsphere
-examples/provider-components/provider-components*.yaml
-test/
+tilt-settings.json
+tilt-settings.yaml
+tilt.d/
+Tiltfile
```
Comment (sbueringer) [inline .dockerignore:17]: There is no titlfile in this repo (probably applies to most of the tilt files here)

File: .dockerignore [hunk 3]
```diff
@@ -1,22 +1,39 @@
...
+# ignores changes to test-only code to avoid extra rebuilds
+test/e2e/**
+
+# We want to ignore any frequently modified files to avoid cache-busting the COPY ./ ./
+# Binaries for programs and plugins
+**/*.exe
+**/*.dll
+**/*.so
+**/*.dylib
+cmd/clusterctl/clusterctl/**
```
Comment (sbueringer) [inline .dockerignore:34]:
```suggestion
```

File: Makefile [hunk 4]
```diff
@@ -488,15 +489,17 @@ DOCKER_BUILD_MODIFY_MANIFESTS ?= true
...
     fi
 
 .PHONY: docker-build-vcsim
 docker-build-vcsim: docker-pull-prerequisites ## Build the docker image for vcsim controller manager
-	DOCKER_BUILDKIT=1 docker build --platform linux/$(ARCH) --build-arg GOLANG_VERSION=$(GO_CONTAINER_IMAGE) --build-arg goproxy=$(GOPROXY) --build-arg ldflags="$(LDFLAGS)" . -t $(VCSIM_CONTROLLER_IMG)-$(ARCH):$(TAG)
+## reads Dockerfile from stdin to avoid an incorrectly cached Dockerfile (https://github.com/moby/buildkit/issues/1368)
+	cat $(VCSIM_DIR)/Dockerfile | DOCKER_BUILDKIT=1 docker build --build-arg builder_image=$(GO_CONTAINER_IMAGE) --build-arg goproxy=$(GOPROXY) --build-arg ARCH=$(ARCH) --build-arg ldflags="$(LDFLAGS)" . -t $(VCSIM_CONTROLLER_IMG)-$(ARCH):$(TAG) --file -
 	@if [ "${DOCKER_BUILD_MODIFY_MANIFESTS}" = "true" ]; then \
   		$(MAKE) set-manifest-image MANIFEST_IMG=$(VCSIM_CONTROLLER_IMG)-$(ARCH) MANIFEST_TAG=$(TAG) TARGET_RESOURCE="./test/infrastructure/vcsim/config/default/manager_image_patch.yaml"; \
 		$(MAKE) set-manifest-pull-policy TARGET_RESOURCE="./test/infrastructure/vcsim/config/default/manager_pull_policy.yaml"; \
```
Comment (sbueringer) [inline Makefile:505]: Maybe use VCSIM_DIR here as well

**Comment by sbueringer** [comment]
/lgtm

Feel free to squash

/assign @chrischdi

**Review by chrischdi — COMMENTED** [review]
/approve

**Pull request merged**